### PR TITLE
feat: Add an option to set bogus resolution constraints for getDisplayMedia.

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: modules/RTC/ScreenObtainer.js [hunk 1]
```diff
@@ -215,14 +215,26 @@ const ScreenObtainer = {
...
 
         const { desktopSharingFrameRate } = this.options;
-        const video = typeof desktopSharingFrameRate === 'object' ? { frameRate: desktopSharingFrameRate } : true;
+        const { setScreenSharingResolutionConstraints } = this.options.testing;
+        const video = {};
+
+        if (typeof desktopSharingFrameRate === 'object') {
+            video.frameRate = desktopSharingFrameRate;
+        }
+        if (setScreenSharingResolutionConstraints) {
```
Comment (jallamsetty1) [inline modules/RTC/ScreenObtainer.js:224]: Don't we want to check if browser is chromium based ?

File: modules/RTC/ScreenObtainer.js [hunk 2]
```diff
@@ -215,14 +215,26 @@ const ScreenObtainer = {
         }
 
         const { desktopSharingFrameRate } = this.options;
-        const video = typeof desktopSharingFrameRate === 'object' ? { frameRate: desktopSharingFrameRate } : true;
+        const { setScreenSharingResolutionConstraints } = this.options.testing;
```
Comment (jallamsetty1) [inline modules/RTC/ScreenObtainer.js:218]: JS will throw an error here if `this.options.testing` is undefined.

**Review by jallamsetty1 — DISMISSED** [review]

**Review by fippo — APPROVED** [review]

File: modules/RTC/ScreenObtainer.js [hunk 3]
```diff
@@ -215,14 +215,27 @@ const ScreenObtainer = {
...
+        }
+
         const audio = this._getAudioConstraints();
 
         // At the time of this writing 'min' constraint for fps is not supported by getDisplayMedia.
         video.frameRate && delete video.frameRate.min;
 
         const constraints = {
-            video,
+            video: video || true,
```
Comment (saghul) [inline modules/RTC/ScreenObtainer.js:238]: This is a noop since video is always defined now.

**Review by saghul — COMMENTED** [review]
One nit.

File: modules/RTC/ScreenObtainer.js [hunk 4]
```diff
@@ -215,14 +215,31 @@ const ScreenObtainer = {
...
         // At the time of this writing 'min' constraint for fps is not supported by getDisplayMedia.
         video.frameRate && delete video.frameRate.min;
 
+        if (Object.keys(video).length === 0) {
+            video = true;
+        }
+
         const constraints = {
-            video,
+            video: video,
```
Comment (saghul) [inline modules/RTC/ScreenObtainer.js:242]: You can now remove this change then.

File: modules/RTC/ScreenObtainer.js [hunk 5]
```diff
@@ -215,14 +215,31 @@ const ScreenObtainer = {
...
+            video.width = 99999;
+        }
+
         const audio = this._getAudioConstraints();
 
         // At the time of this writing 'min' constraint for fps is not supported by getDisplayMedia.
         video.frameRate && delete video.frameRate.min;
 
+        if (Object.keys(video).length === 0) {
+            video = true;
```
Comment (saghul) [inline modules/RTC/ScreenObtainer.js:238]: You best to make video a let now.

Comment (bgrozev) [reply modules/RTC/ScreenObtainer.js:238]: Sorry, I'm using a new editor and not paying enough attention apparently 🤦

**Review by saghul — APPROVED** [review]

**Pull request merged**